### ISSUE TITLE
False positive `unused-variable` for `unittest` test classes

### ISSUE THREAD

**Issue by DervishD** [PyCQA/pylint#7032]
### Bug description

`pylint` is giving `unused-variable` for the class `TestSomething` in the code below, since it is an `unittest` test case, which is not intended to be instantiated and run directly, but by using `unittest.main()`.

And yes, the caller *CAN* instantiate the class, add it to a test suite, etc. and *THEN* those identifiers will be used, but IMHO the principal use case of `unittest` is to not instantiate the test case classes.

I know that I can disable the warning for those classes (I'm already doing that in my code), but I'm reporting this just in case it is desirable to disable the warning by default in that particular context.

I absolutely understand that this is only an exception but since I think it is a common code pattern, I prefer to report it just in case it had been missed.

Thanks in advance :)

```
import unittest

class TestSomething(unittest.TestCase):
    """Class for testing something."""
    def test_whatever(self):
        """Useless test, but…"""
        self.assertEqual(1, 1)

def main():
    # The code below will automagically instantiate TestSomething and will run test_whatever().
    unittest.main()
```
    

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:3:0: W0612 (unused-variable) Unused variable 'TestSomething'
```

### Expected behavior

IMHO, this warning should be disabled by default for this particular use pattern, as it is common when using module 'unittest'

### Pylint version

```shell
pylint 2.14.2
astroid 2.11.6
Python 3.10.4 (tags/v3.10.4:9d38120, Mar 23 2022, 23:13:41) [MSC v.1929 64 bit (AMD64)]
```

### OS / Environment

Windows 10 Home

**Comment by Pierre-Sassoulas** [comment]
Hmm, it kinda makes sense and would actually be a false positive... But
1) I can't reproduce on main, are you sure you're using 2.14.2 ?
2) We never had a report in several years regarding this. I think it's because you don't need to have this ``main()`` function in order for unittest to automagically launch all your test classes with a single command. (It could also be python 3.10.) What happen if you remove the ``main()`` ?

**Comment by jacobtylerwalls** [comment]
Reproduces with `--allow-global-unused-variables=no`.

@DervishD is that in your config? If so, you could just set it to `yes`, our current default.

**Comment by DervishD** [comment]
@Pierre-Sassoulas yes, I'm using 2.14.2, and happened even without `main()`, the problem lied elsewhere. To wit, my stupidity.
@jacobtylerwalls no, that was not in my config, I checked and even reproduced with a default config. It was **in the project's config** 🤦‍♂️

I'm quite embarrassed, please excuse me for the noise, and @jacobtylerwalls thanks a lot for pointing this, otherwise I would have NEVER thought that the project had a local config. I'm very sorry.
